Makefile

```diff
@@ -577,6 +577,11 @@ ifeq ($(dot-config),1)
 # oldconfig if changes are detected.
 -include include/config/auto.conf.cmd
 
+# Read in DTS derived configuration, if it exists
+ifneq ($(ARCH),)
```

Comment:
Short story: this is to circumvent the multiple updates of the config file that are done as a side effect of the implicit rule for the include file.  This might need a large paragraph to explain why.